### ISSUE TITLE
FE8 Game frozen when Artur attacks Monster Ch.4

### ISSUE THREAD

**Issue by JacobKleinsorge** [lushen124/Universal-FE-Randomizer#319]
Artur changed to a dancer. He shows up, attacks the monster from range with light and does 0 damage to it. Then the game freezes.

Workaround I found was to press start to skip the whole cutscene. That way the monster dies and I can just keep playing from the actual start of the mission.

**Comment by lushen124** [comment]
Yeah, scripted fights tend to cause issues. I have a mechanism to handle this in most cases, but I forgot about this skippable scripted fight. Normally I would lock Artur as needing range (much like Erk in FE7's Lyn Mode), but on the other hand it is skippable, so whether it's worth it to restrict his class to non-dancer, non-healer, and non-swordlock is an open question. My initial thought is to just apply the same logic as I did for Erk.
